constraint/pkg/client/client.go

```diff
@@ -415,7 +416,22 @@ func (c *Client) AddData(ctx context.Context, data interface{}) (*types.Response
 			}
 		}
 
-		err = c.driver.AddData(ctx, name, key, processedData)
+		// Round trip data to force untyped JSON, as drivers are not type-aware
+		bytes, err := json.Marshal(processedData)
```

Comment:
For my own knowledge, why does doing round trip here improves performance compare to in storage?

Reply:
Storage acquires a mutex, so a round trip there blocks all Rego evaluation. Round trips here can be done in parallel without blocking Rego evaluation.

TL;DR less CPU time spent with a mutex